modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java

```diff
@@ -1426,7 +1436,18 @@ private JdbcThinTcpIo cliIo(List<UUID> nodeIds) {
             }
         }
 
-        return io != null ? io : iosArr[RND.nextInt(iosArr.length)];
+        return io != null ? io : randomIo();
+    }
+
+    /**
+     * @return random tcpIo
+     */
+    private JdbcThinTcpIo randomIo() {
+        UUID randomUUID = UUID.randomUUID();
```

Comment:
`UUID.randomUUID` is a static method and uses a globally shared random number generator (RNG). It can bring an unexpected performance impact on random UUID generation. Can we replace it with private RNG (borrowing ideas from `randomUUID` implementation)?

Reply:
Yep, benchmarks revealed that there's a serious impact of "out-of-scope" random UUID generator. Whats more, `Random` is much faster than `SecureRandom` and cause we don't need cryptographically strong pseudo random number generator in this particular case, it's perfectly fine to use custom randomUUID() with simple `Random` inside. Fixed.